### ISSUE TITLE
Loading last checkpoint and versioned last checkpoint

### ISSUE THREAD

**Issue by agitter** [gitter-lab/metl#12]
The Lightning [ModelCheckpoint](https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.callbacks.ModelCheckpoint.html) has an argument `save_top_k`
> save_top_k (int) – if save_top_k == k, the best k models according to the quantity monitored will be saved. If save_top_k == 0, no models are saved. If save_top_k == -1, all models are saved. Please note that the monitors are checked every every_n_epochs epochs. If save_top_k >= 2 and the callback is called multiple times inside an epoch, and the filename remains unchanged, the name of the saved file will be appended with a version count starting with v1 to avoid collisions unless enable_version_counter is set to False. The version counter is unrelated to the top-k ranking of the checkpoint, and we recommend formatting the filename to include the monitored metric to avoid collisions.

METL training sets `save_top_k=-1` here
https://github.com/gitter-lab/metl/blob/0d8129e4b4b5e8e77302d046d1ed7c23496a968e/code/train_source_model.py#L267-L272

We do not expect to have versioned "last" checkpoints, but they were observed under certain conditions while training in AWS. That caused a problem when loading checkpoints with filenames like `last-v3.ckpt` using the code here
https://github.com/gitter-lab/metl/blob/0d8129e4b4b5e8e77302d046d1ed7c23496a968e/code/train_source_model.py#L94-L101

that looks for `last.ckpt`.
